cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/aws-mc-common/aws-orchestrator/AwsOrchestrator.service.tsx

```diff
@@ -46,29 +40,26 @@ export const nodeProfiles: NodeProfileType[] = [
 export class AwsOrchestrator {
     static async initOsImages(props: AwsOrchestratorProps) {
         const { awsState, awsDispatch, setErrorObject, errorObject } = props;
-        try {
-            const osImages = await AwsService.getAwsosImages(awsState[STORE_SECTION_FORM].REGION);
-            saveCurrentResourceData(awsDispatch, RESOURCE.AWS_ADD_RESOURCES, AWS_FIELDS.OS_IMAGE, osImages);
-            setDefaultOsImage(awsDispatch, osImages);
-            setErrorObject(removeErrorInfo(errorObject, AWS_FIELDS.OS_IMAGE));
-        } catch (e) {
-            clearPreviousResourceData(awsDispatch, RESOURCE.AWS_ADD_RESOURCES, AWS_FIELDS.OS_IMAGE);
-            setErrorObject(addErrorInfo(errorObject, e, AWS_FIELDS.OS_IMAGE));
-        }
+        await DefaultOrchestrator.initResources<AWSVirtualMachine>({
+            resourceName: AWS_FIELDS.OS_IMAGE,
+            dispatch: awsDispatch,
+            errorObject,
+            setErrorObject,
+            fetcher: () => AwsService.getAwsosImages(awsState[STORE_SECTION_FORM].REGION),
+            fxnSelectDefault: AwsDefaults.selectDefaultOsImage,
+        });
     }
 
-    static async initEC2KeyPairs(props: AwsOrchestratorProps, setKeyPairs: (keyPairs: AWSKeyPair[]) => void) {
+    static async initEC2KeyPairs(props: AwsOrchestratorProps) {
         const { awsDispatch, setErrorObject, errorObject } = props;
-        try {
-            const keyPairs = await AwsService.getAwsKeyPairs();
-            saveCurrentResourceData(awsDispatch, RESOURCE.AWS_ADD_RESOURCES, AWS_FIELDS.EC2_KEY_PAIR, keyPairs);
-            setDefaultEC2KeyPair(awsDispatch, keyPairs);
-            setErrorObject(removeErrorInfo(errorObject, AWS_FIELDS.OS_IMAGE));
-            setKeyPairs(keyPairs);
-        } catch (e) {
-            clearPreviousResourceData(awsDispatch, RESOURCE.AWS_ADD_RESOURCES, AWS_FIELDS.EC2_KEY_PAIR);
-            setErrorObject(addErrorInfo(errorObject, e, AWS_FIELDS.EC2_KEY_PAIR));
-        }
+        await DefaultOrchestrator.initResources<AWSKeyPair>({
+            resourceName: AWS_FIELDS.EC2_KEY_PAIR,
+            dispatch: awsDispatch,
+            errorObject,
+            setErrorObject,
+            fetcher: () => AwsService.getAwsKeyPairs(),
+            fxnSelectDefault: AwsDefaults.selectDefaultEC2KeyPairs,
+        });
     }
 
     static async initNodeProfile(props: AwsOrchestratorProps) {
```

Comment:
Do we want to use `DefaultOrchestrator` for `initNodeProfile` ?

Reply:
I think AWS will have a different strategy than vSphere, so I'd suggest we leave it here for now.